### ISSUE TITLE
Test cleanup on closing debugger (3.1pr3)

### ISSUE THREAD

**Issue by LisaAlmarode** [ericwinger/Jade#768]
I have a setup for testing Plantis similar to the setup that James put into the bug system, which uses SessionTemps and a current and a default to setup test data and then cleanup afterwards.  I run these tests a great deal using GBS. A critical thing is that tearDown is always invoked so after tests are done I can see my production data, not my test data.  This is reliable in GBS.

In Jadeite, if I do a run-all where I have a breakpoint in a method so it brings up the debugger, and close the debugger using the windows upper left corner X (as is my usual habit) the system is left with test data.  I can logout and I suppose manually remove the key from SessionTemps but I'd rather tearDown was reliably invoked.

**Comment by ericwinger** [comment]
I think this might be a server bug in 3.2.15.  If you put halts in this method as I've done you'll see the test start, but when the debugger is closed, the ensure block is not performed. 

Jadeite terminates the process using `RowanDebuggerService>>terminateProcess:` which sends `terminate` to the `GsProcess`.

What I don't know if this is expected behavior. My suspicion is that it's related to this server bug reported against a later server version. 
https://kermit.gemtalksystems.com/bug?bug=48271

```
runCase
	| tornDown |
	self halt. 
	self resources do: [:each | each availableFor: self].
	[ self setUp.
	  self performTest
        ] ensure: [
	  tornDown ifNil:[ 
		tornDown := true . self halt. 
		self tearDown
	  ]
        ]
```

**Comment by LisaAlmarode** [comment]
This appears to be fixed as of 3.2.12 and 3.7 (in which bug 48271 is fixed).
